lib/replicator.js

```diff
@@ -111,7 +106,13 @@ class BlockRequest extends Attachable {
   }
 
   _unref () {
-    if (this.inflight.length > 0) return
+    if (this.inflight.length > 0) {
+      for (const req of this.inflight) {
+        req.peer._cancelRequest(req.id)
+      }
+      return
```

Comment:
i ... think ... you wanna remove this return, otherwise no one removes it from the tracker